Rpackage/R/mmit.cv.R

```diff
@@ -0,0 +1,104 @@
+#' The Cross Validation of Max Margin Interval Tree
+#'
+#' Performing grid search to select the best parameters via cross validation on the  a regression tree for censored data.
+#' 
+#' @param target.mat The response variable of the model
+#' @param feature.mat a data frame containing the feature variables in the model.
+#' @param param_grid the list of paramaters
+#' @param n_folds The number of folds
+#' @param scorer The Loss calculation function 
+#' @param n_cpu The number of cores to register for parallel programing of the code, default value is 1 and n_cpu = -1 to select all cores.
+#' @param pruning Boolean whether pruning is to be done or not.
+#' 
+#' @return The list consist of best score, best tree, best parameters and list of all parameter values with cross validation score . 
+#' 
+#' @author Toby Dylan Hocking, Alexandre Drouin, Torsten Hothorn, Parismita Das
+#' 
+#' @examples
+#' library(mmit)
+#' target.mat <- rbind(
+#'   c(0,1), c(0,1), c(0,1),
+#'   c(2,3), c(2,3), c(2,3))
+#' 
+#' feature.mat <- rbind(
+#'   c(1,0,0), c(1,1,0), c(1,2,0),
+#'   c(1,3,0), c(1,4,0), c(1,5,0))
+#' 
+#' colnames(feature.mat) <- c("a", "b", "c")
+#' feature.mat <- data.frame(feature.mat)
+#' 
+#' param_grid <- NULL
+#' param_grid$max_depth <- c(Inf, 4, 3)
+#' param_grid$margin <- c(2, 3, 5)
+#' param_grid$min_sample <- c(2, 5, 10)
+#' param_grid$loss <- c("hinge")
+#' 
+#' result <- mmit.cv(target.mat, feature.mat, param_grid, scorer = mse)
+#' 
+#' @export
+mmit.cv <- structure(function(target.mat, feature.mat, 
+                              param_grid, n_folds = 3,
+                              scorer = NULL, n_cpu = 1, 
+                              pruning = TRUE){
+  
+  ### add default value to parameters
+  if(is.null(param_grid$max_depth)) param_grid$max_depth <- Inf
+  if(is.null(param_grid$margin)) param_grid$margin <- 0.0
+  if(is.null(param_grid$min_sample)) param_grid$min_sample <- 0.0
+  if(is.null(param_grid$loss)) param_grid$loss <- "hinge"
+  
+  ### check for unwanted parameters
+  assert_that(length(param_grid) <= 4, msg = "unexpected parameters as arguement")
```

Comment:
This condition does not work. I discovered this by mistake. 

There is a very weird feature in R that allows to match elements in list with partial names. For instance, if the list contains an entry with name "alex", then you can get it with mylist$al and also mylist$ale, and also mylist$alex.

So basically, when I created a param_grid with min_samples instead of min_sample, it was passing validation properly, but failing later in the code.

The solution is to replace mylist$alex with mylist[[alex]] over here.

Reply:
done